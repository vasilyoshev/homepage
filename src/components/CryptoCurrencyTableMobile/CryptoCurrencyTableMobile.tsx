import React from 'react';
import styles from './CryptoCurrencyTableMobile.module.scss';
import { Card, CardContent } from '@material-ui/core';

export const CryptoCurrencyTableMobile: React.FC<any> = ({ coins }: any) => {
  return (
    <>
      {!!coins.length && coins?.map((coin: any) => (
        <Card key={coin.id}>
          <CardContent>
            <div>{coin.name}</div>
            <div>{coin.image}</div>
            <div>{coin.symbol}</div>
            <div>{coin.market_cap}</div>
            <div>{coin.current_price}</div>
            <div>{coin.price_change_percentage_24h}</div>
          </CardContent>
        </Card>
      ))}
    </>
  );
};
