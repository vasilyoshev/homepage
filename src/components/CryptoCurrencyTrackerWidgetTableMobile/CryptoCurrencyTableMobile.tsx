import React from 'react';
import { Card, CardContent } from '@material-ui/core';
import { CryptoCurrencyTrackerWidgetResponseTableProps } from 'interfaces';
import styles from './CryptoCurrencyTableMobile.module.scss';

export const CryptoCurrencyTableMobile: React.FC<CryptoCurrencyTrackerWidgetResponseTableProps> = ({ coins }) => {
  return (
    <>
      {!!coins.length && coins.map((coin) => (
        <Card className={styles.card} key={coin.id}>
          <CardContent>
            <div className={styles.coinName}>
              {coin.name}
              <img
                className={styles.icon}
                src={coin.image}
                alt="crypto" />
            </div>
            <div>Symbol: {coin.symbol.toUpperCase()}</div>
            <div>
              Price: <p className={styles.coinPrice}>${coin.current_price.toFixed(2)}</p>
            </div>
            <div>Price change in last 24h:
              {coin.price_change_percentage_24h < 0 ? (
                <p className={styles.coinPercentRed}>-{coin.price_change_percentage_24h.toFixed(2)}%</p>
              ) : (
                <p className={styles.coinPercentGreen}>+{coin.price_change_percentage_24h.toFixed(2)}%</p>
              )}
            </div>
            <div>Market Cap: ${coin.market_cap.toFixed(2)}</div>
          </CardContent>
        </Card>
      ))}
    </>
  );
};
