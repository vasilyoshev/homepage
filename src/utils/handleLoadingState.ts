import axios from 'axios';
import { decrementLoadingCount, incrementLoadingCount } from 'slices';
import store from 'store';

export const handleLoadingState = (): void => {
  axios.interceptors.response.use(
    (response) => {
      setTimeout(() => store.dispatch(decrementLoadingCount()), 200);

      return response;
    },
    () => {
      store.dispatch(decrementLoadingCount());
    },
  );

  axios.interceptors.request.use((request) => {
    store.dispatch(incrementLoadingCount());

    return request;
  });
};
