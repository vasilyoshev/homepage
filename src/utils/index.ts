export * from './handleLoadingState';
export * from './testUtils';
