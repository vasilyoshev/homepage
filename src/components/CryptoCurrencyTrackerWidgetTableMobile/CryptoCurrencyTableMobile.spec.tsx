import { render } from 'utils';
import { CryptoCurrencyTableMobile } from './CryptoCurrencyTableMobile';

describe('CryptoCurrencyTableMobile', () => {
  it('renders successfully', () => {
    const { asFragment } = render(<CryptoCurrencyTableMobile coins={[]} />);
    
    expect(asFragment()).toMatchSnapshot();
  });
});
