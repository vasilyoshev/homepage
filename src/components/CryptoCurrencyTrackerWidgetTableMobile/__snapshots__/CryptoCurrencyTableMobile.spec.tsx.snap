// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CryptoCurrencyTableMobile renders successfully 1`] = `<DocumentFragment />`;
